### ISSUE TITLE
ConnectionError to Redis in 2.2.0

### ISSUE THREAD

**Issue by EnotYoyo** [alex-oleshkevich/starsessions#76]
Hi,

First of all, thank you for maintaining and developing this library, it has been very useful for us!

I recently upgraded to `starsessions` version `2.2.0` and started experiencing intermittent issues with connections to Redis, which didn't occur with version `2.1.3`. After reviewing the diff between versions, it seems the issue might be related to changes in the connection handling mechanism, possibly involving the use of connections via a context manager. Below is the stack trace I encountered:
```python
Traceback (most recent call last):
  File "/path/to/virtualenv/site-packages/uvicorn/protocols/http/httptools_impl.py", line 401, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/path/to/virtualenv/site-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__
    return await self.app(scope, receive, send)
  File "/path/to/virtualenv/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/path/to/virtualenv/site-packages/starlette/applications.py", line 113, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/path/to/virtualenv/site-packages/starlette/middleware/errors.py", line 187, in __call__
    raise exc
  File "/path/to/virtualenv/site-packages/starlette/middleware/errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
  File "/path/to/virtualenv/site-packages/prometheus_fastapi_instrumentator/middleware.py", line 174, in __call__
    raise exc
  File "/path/to/virtualenv/site-packages/prometheus_fastapi_instrumentator/middleware.py", line 172, in __call__
    await self.app(scope, receive, send_wrapper)
  File "/path/to/virtualenv/site-packages/starlette/middleware/base.py", line 185, in __call__
    with collapse_excgroups():
  File "/path/to/python/lib/contextlib.py", line 158, in __exit__
    self.gen.throw(value)
  File "/path/to/virtualenv/site-packages/starlette/_utils.py", line 82, in collapse_excgroups
    raise exc
  File "/path/to/virtualenv/site-packages/starlette/middleware/base.py", line 187, in __call__
    response = await self.dispatch_func(request, call_next)
  File "/path/to/project/app.py", line 119, in set_log_request_id
    response: Response = await call_next(request)
  File "/path/to/virtualenv/site-packages/starlette/middleware/base.py", line 163, in call_next
    raise app_exc
  File "/path/to/virtualenv/site-packages/starlette/middleware/base.py", line 149, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "/path/to/virtualenv/site-packages/starsessions/middleware.py", line 147, in __call__
    await self.app(scope, receive, send_wrapper)
  File "/path/to/virtualenv/site-packages/starsessions/middleware.py", line 166, in __call__
    await load_session(connection)
  File "/path/to/virtualenv/site-packages/starsessions/session.py", line 49, in load_session
    await get_session_handler(connection).load()
  File "/path/to/virtualenv/site-packages/starsessions/session.py", line 111, in load
    await self.store.read(
  File "/path/to/virtualenv/site-packages/starsessions/stores/redis.py", line 56, in read
    value = await client.get(self.prefix(session_id))
  File "/path/to/virtualenv/site-packages/redis/asyncio/client.py", line 616, in execute_command
    return await conn.retry.call_with_retry(
  File "/path/to/virtualenv/site-packages/redis/asyncio/retry.py", line 62, in call_with_retry
    await fail(error)
  File "/path/to/virtualenv/site-packages/redis/asyncio/client.py", line 603, in _disconnect_raise
    raise error
  File "/path/to/virtualenv/site-packages/redis/asyncio/retry.py", line 59, in call_with_retry
    return await do()
  File "/path/to/virtualenv/site-packages/redis/asyncio/client.py", line 590, in _send_command_parse_response
    return await self.parse_response(conn, command_name, **options)
  File "/path/to/virtualenv/site-packages/redis/asyncio/client.py", line 637, in parse_response
    response = await connection.read_response()
  File "/path/to/virtualenv/site-packages/redis/asyncio/connection.py", line 543, in read_response
    response = await self._parser.read_response(
  File "/path/to/virtualenv/site-packages/redis/_parsers/hiredis.py", line 211, in read_response
    await self.read_from_socket()
  File "/path/to/virtualenv/site-packages/redis/_parsers/hiredis.py", line 191, in read_from_socket
    raise ConnectionError(SERVER_CLOSED_CONNECTION_ERROR) from None
redis.exceptions.ConnectionError: Connection closed by server.

```
In my setup, this issue occurs periodically, but the same configuration worked well in version 2.1.3. I use the last version of Redis in Docker.

**Comment by alex-oleshkevich** [comment]
Thanks, I see the problem. Need to think about this.
Async Redis requires explicit connection disconnect - https://redis-py.readthedocs.io/en/latest/examples/asyncio_examples.html

No idea at the moment how to handle this without breaking the API.

**Comment by alex-oleshkevich** [comment]
I have a fix in 2.2.1
